snappiershot/serializers/utils.py

```diff
@@ -21,13 +45,15 @@ def default_encode_value(value: Any) -> JsonType:
         encoded_dict = dict()
         for key, item in value.items():
             try:
-                encoded_dict[key] = default_encode_value(item)
-            except ValueError:
-                warnings.warn(
-                    f"Cannot serialize this value: {item} -- "
-                    f"Skipping over this (key, value) pair. ",
-                    SnappierShotWarning,
-                )
+                encoded_dict[key] = default_encode_value(item, context)
+            except (ValueError, RecursionError) as err:
```

Comment:
This fine, but, as I'm sure you know, you can have multiple `except`  catches for one `try` block (with different exception types)

Reply:
[out of scope] this bit of code/logic is repeated below (around line 69). should this be moved to a helper function?

Reply:
I think it's OK for now. We can revisit this when we work out how the CustomErrors are incorporated.